scope.go

```diff
@@ -1324,7 +1324,7 @@ func (scope *Scope) autoIndex() *Scope {
 }
 
 func (scope *Scope) getColumnAsArray(columns []string, values ...interface{}) (results [][]interface{}) {
-	resultMap := make(map[string][]interface{})
+	resultMap := make(map[string]bool)
```

Comment:
use struct{} instead of bool, you are not using that value anywhere

Reply:
Yes the map's value is not used. Using this(i.e. resultMap) as a set.
Didn't know about struct{} thing. Just asking is that common?

I will make the changes. But it does not seem this will get into master. [Ref](https://github.com/jinzhu/gorm/issues/2886)